lib/pages/athlete/bloc/AthletePageBloc.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:ax_dapp/pages/athlete/models/AthletePageEvent.dart';
+import 'package:ax_dapp/pages/athlete/models/AthletePageState.dart';
+import 'package:ax_dapp/repositories/MlbRepo.dart';
+import 'package:ax_dapp/service/athleteModels/mlb/MLBPAthleteStats.dart';
+import 'package:ax_dapp/util/BlocStatus.dart';
+import 'package:ax_dapp/util/chart/extensions/graphData.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:intl/intl.dart';
+
+class AthletePageBloc extends Bloc<AthletePageEvent, AthletePageState> {
+  final MLBRepo repo;
+
+  AthletePageBloc({required this.repo}) : super(AthletePageState.initial()) {
+    on<OnPageRefresh>(_mapPageRefreshEventToState);
+    on<OnGraphRefresh>(_mapGraphRefreshEventToState);
+  }
+
+  void _mapPageRefreshEventToState(
+      OnPageRefresh event, Emitter<AthletePageState> emit) async {
+    try {
+      print("load page stats");
+      final int playerId = event.playerId;
+      emit(state.copyWith(status: BlocStatus.loading));
+      final MLBAthleteStats stats = await repo.getPlayerStatsHistory(
+          playerId, "2022-06-01", "2022-06-30");
```

Comment:
Lets not hardcode the dates in. We can try to get the current date and the older date and then pass them in here.

Reply:
done, as for the older date there is no way to get that